module/solutions/helper.js

```diff
@@ -935,20 +945,23 @@ module.exports = class SolutionsHelper {
 	static queryBasedOnRoleAndLocation(data, type = '') {
 		return new Promise(async (resolve, reject) => {
 			try {
-				let entities = []
+				let registryIds = []
 				let entityTypes = []
 				let filterQuery = {
 					isReusable: false,
 					isDeleted: false,
 				}
+				Object.keys(_.omit(data, ['role', 'filter', 'factor'])).forEach((key) => {
+					data[key] = data[key].split(',')
+				})
 
 				// If validate entity set to ON . strict scoping should be applied
 				if (validateEntity !== CONSTANTS.common.OFF) {
-					Object.keys(_.omit(data, ['filter', 'role', 'factors', 'type'])).forEach((requestedDataKey) => {
-						if (requestedDataKey == 'entities') entities.push(...data[requestedDataKey])
-						if (requestedDataKey == 'entityType') entityTypes.push(data[requestedDataKey])
+					Object.keys(_.omit(data, ['filter', 'role', 'factors'])).forEach((requestedDataKey) => {
```

Comment:
@Prajwal17Tunerlabs type is missing here also

Reply:
type keyword added.